File: Examples/test-suite/ocaml/template_methods_runme.ml
open Swig
open Template_methods

let _ =
  let num = C_float 1. in
  assert (_convolve1Bool '() as int = 0);
  assert (_convolve1Bool '(true) = C_void);
  assert (_convolve2Float '() as int = 0);
  assert (_convolve3FloatRenamed '(num) = C_void);
  assert (_convolve4Float '() as int = 0);
  assert (_convolve4FloatRenamed '(num) = C_void);
  assert (_convolve5FloatRenamed '() as int = 0);
  assert (_convolve5FloatRenamed '(num) = C_void);

  let k = new_Klass '() in
  assert (k -> KlassTMethodBoolRenamed (true) as bool = true);
  assert (k -> KlassTMethodBool () = C_void);
  assert (_Klass_KlassStaticTMethodBoolRenamed '(true) as bool = true);
  assert (_Klass_KlassStaticTMethodBool '() = C_void);

  let cp = new_ComponentProperties '() in
  assert (cp -> adda ("key1", "val1", "key2", 22.2) = C_void);
  assert (cp -> adda ("key1", "val1", "key2", "val2", "key3", "val3") = C_void);
  assert (cp -> adda ("key1", 1, "key2", 2, "key3", 3) = C_void)
;;

File: Examples/test-suite/lua/template_methods_runme.lua
local v=require("template_methods")

local num = 1.1

-- Global templated functions
local i = v.convolve1Bool()
v.convolve1Bool(true)
i = v.convolve2Float()
v.convolve3FloatRenamed(num)
i = v.convolve4Float()
v.convolve4FloatRenamed(num)
i = v.convolve5FloatRenamed()
v.convolve5FloatRenamed(num)

-- Static templated methods
local k = v.Klass()
local b = k:KlassTMethodBoolRenamed(true)
k:KlassTMethodBool()
b = v.Klass.KlassStaticTMethodBoolRenamed(true)
v.Klass.KlassStaticTMethodBool()

local cp = v.ComponentProperties()
cp:adda("key1", "val1", "key2", 22.2)
cp:adda("key1", "val1", "key2", "val2", "key3", "val3")
cp:adda("key1", 1, "key2", 2, "key3", 3)
